evm/v0.5/contracts/dev/PrepaidAggregator.sol

```diff
@@ -0,0 +1,429 @@
+   */
+  function updateFutureRounds(
+    uint128 _newPaymentAmount,
+    uint64 _minAnswers,
+    uint64 _maxAnswers,
+    uint64 _restartDelay
+  )
+    public
+    onlyOwner()
+    onlyValidRange(_minAnswers, _maxAnswers, _restartDelay)
+  {
+    paymentAmount = _newPaymentAmount;
+    minAnswerCount = _minAnswers;
+    maxAnswerCount = _maxAnswers;
+    restartDelay = _restartDelay;
+
+    emit RoundDetailsUpdated(
+      paymentAmount,
+      _minAnswers,
+      _maxAnswers,
+      _restartDelay
+    );
+  }
+
+  /**
+   * @notice recalculate the amount of LINK available for payouts
+   */
+  function updateAvailableFunds()
+    public
+  {
+    uint256 available = LINK.balanceOf(address(this)).sub(allocatedFunds);
+    availableFunds = uint128(available);
+    emit AvailableFundsUpdated(available);
+  }
+
+  /**
+   * @notice query the available amount of LINK for an oracle to withdraw
+   */
+  function withdrawable()
+    external
+    view
+    returns (uint256)
+  {
+    return uint256(oracles[msg.sender].withdrawable);
+  }
+
+  /**
+   * @notice get the most recently reported answer
+   */
+  function currentAnswer()
+    external
+    view
+    returns (int256)
+  {
+    return getAnswer(latestRound);
+  }
+
+  /**
+   * @notice get the last updated at timestamp
+   */
+  function updatedTimestamp()
+    external
+    view
+    returns (uint256)
+  {
+    return getUpdatedTimestamp(latestRound);
+  }
+
+  /**
+   * @notice get past rounds answers
+   * @param _id the round number to retrieve the answer for
+   */
+  function getAnswer(uint128 _id)
+    public
+    view
+    returns (int256)
+  {
+    return rounds[_id].answer;
+  }
+
+  /**
+   * @notice get timestamp when an answer was last updated
+   * @param _id the round number to retrieve the updated timestamp for
+   */
+  function getUpdatedTimestamp(uint128 _id)
+    public
+    view
+    returns (uint256)
+  {
+    return rounds[_id].updatedTimestamp;
+  }
+
+  /**
+   * @notice transfers the oracle's LINK to another address
+   * @param _recipient is the address to send the LINK to
+   * @param _amount is the amount of LINK to send
+   */
+  function withdraw(address _recipient, uint256 _amount)
+    external
+  {
+    uint128 amount = uint128(_amount);
+    uint128 available = oracles[msg.sender].withdrawable;
+    require(available >= amount, "Insufficient balance");
+
+    oracles[msg.sender].withdrawable = available.sub(amount);
+    allocatedFunds = allocatedFunds.sub(amount);
+
+    assert(LINK.transfer(_recipient, _amount));
+  }
+
+  /**
+   * @notice transfers the owner's LINK to another address
+   * @param _recipient is the address to send the LINK to
+   * @param _amount is the amount of LINK to send
+   */
+  function withdrawFunds(address _recipient, uint256 _amount)
+    external
+    onlyOwner()
+  {
+    require(availableFunds >= _amount, "Insufficient funds");
+    require(LINK.transfer(_recipient, _amount), "LINK transfer failed");
```

Comment:
Why would it be better? I tend to lean towards require because it doesn't eat all gas. Not that this is an authority, but [this SO answer](https://ethereum.stackexchange.com/questions/15166/difference-between-require-and-assert-and-the-difference-between-revert-and-thro) recommends require for external contract calls.

Reply:
We use assert in function `withdraw`. I don't have a strong opinion either way as long as we're consistent.